src/sentry/search/events/datasets/base.py

```diff
@@ -55,3 +103,29 @@ def _release_filter_converter(self, search_filter: SearchFilter) -> Optional[Whe
         return self.builder._default_filter_converter(
             SearchFilter(search_filter.key, operator, value)
         )
+
+    # Field Aliases
+    def _resolve_project_slug_alias(self, alias: str) -> SelectType:
+        project_ids = {
+            project_id
+            for project_id in self.builder.params.get("project_id", [])
+            if isinstance(project_id, int)
+        }
+
+        # Try to reduce the size of the transform by using any existing conditions on projects
+        # Do not optimize projects list if conditions contain OR operator
+        if not self.builder.has_or_condition and len(self.builder.projects_to_filter) > 0:
+            project_ids &= self.builder.projects_to_filter
+
+        projects = Project.objects.filter(id__in=project_ids).values("slug", "id")
+
+        return Function(
```

Comment:
Going to do this in a separate PR, since it requires us to change the order of operations (we currently resolve orderby after columns) and once I do that this optimization can be used for this Builder & Discover's